app/uk/gov/hmrc/gform/sharedmodel/formtemplate/FormComponentId.scala

```diff
@@ -31,8 +33,17 @@ object FormComponentId {
   implicit val catsEq: Eq[FormComponentId] = Eq.fromUniversalEquals
 
   implicit val vformat: Format[FormComponentId] =
-    ValueClassFormat.vformat("id", FormComponentId.apply, x => JsString(x.value))
+    ValueClassFormat.validatedvformat("id", validate, x => JsString(x.value))
 
   val oformat: OFormat[FormComponentId] = ValueClassFormat.oformat("id", FormComponentId.apply, _.value)
 
+  val idValidationRegex: Regex = """[_a-zA-Z]\w*""".r.unanchored
```

Comment:
This would be better named idValidation, we do not commonly call values "firstNameString", "totalBunsInteger" .

Programmers did once use a convention like that, but it long ago went out of fashion, at least in most languages

Reply:
Also, you don't need the .unanchored here, it is unanchored by default